src/hotspot/share/nmt/vmatree.hpp

```diff
@@ -165,16 +168,37 @@ class VMATree {
         flag[i] = SingleDiff{0, 0};
       }
     }
+    SummaryDiff(int invalid_amount) {
+      for (int i = 0; i < mt_number_of_types; i++) {
+        flag[i] = SingleDiff{invalid_amount, invalid_amount};
+      }
+    }
+
+    bool is_valid(int invalid_amount = -1) {
+      for (int i = 0; i < mt_number_of_types; i++) {
+        if (flag[i].reserve != invalid_amount)
+          return true;
+        if (flag[i].commit != invalid_amount)
+          return true;
+      }
+      return false;
+    }
   };
 
-  SummaryDiff register_mapping(position A, position B, StateType state, const RegionData& metadata);
+ private:
+  SummaryDiff register_mapping(position A, position B, StateType state, const RegionData& metadata, bool copy_flag = false);
 
+ public:
   SummaryDiff reserve_mapping(position from, position sz, const RegionData& metadata) {
```

Comment:
In

`SummaryDiff register_mapping(position A, position B, StateType state, const RegionData& metadata, bool copy_flag = false);`

`position B` is clearly a position of its own, however, in

`SummaryDiff reserve_mapping(position from, position sz, const RegionData& metadata)`

`position sz` is **size** or **delta**. We should rename the parameter to reflect that intended functionality better.

I would probably prefer to see us introduce:

`using position_size = size_t;`

and use as appropriate, instead, of forcing everything to be `position`, to help keep the API usage straight.

Reply:
Hi, I did this change in https://github.com/openjdk/jdk/pull/20994